specification/sections/01-flag-evaluation.md

```diff
@@ -105,6 +105,22 @@ The name is a logical identifier for the client.
 
 Clients may be created in critical code paths, and even per-request in server-side HTTP contexts. Therefore, in keeping with the principle that OpenFeature should never cause abnormal execution of the first party application, this function should never throw. Abnormal execution in initialization should instead occur during provider registration.
 
+#### Requirement 1.1.8
+
+> The `API` **SHOULD** provide functions to set a provider and wait for the `initialize` function to return or throw.
+
+The `API` will ensure that the provider is ready or in error before being able to initializing the client.
+
+```typescript
+// default client
+await Openfeature.setProviderAndWait(myprovider);
+const client = Openfeature.getClient();
+
+// named client
+await Openfeature.setProviderAndWait('client-name', myprovider);
+const client = Openfeature.getClient('client-name');
+```
```

Comment:
In general, for consistency, I've tried to stick to Java in the spec. I created [this](https://github.com/open-feature/spec/issues/187) a while ago to try to make that more consistent. It's up to you, but I'd recommend making this look "Java-ish" and add a comment saying `// this method blocks` or something like that.

Reply:
I've seen a bunch of typescript examples that's why I went for it. I've changed it to Java.